### PR TITLE
US5 - Automatic update not more frequent than 24h

### PR DESCRIPTION
**Pull request by CebotariM**
#Added SharedPrefs
#Edited ForecastActivity

### CODE REVIEW

File: app/src/main/java/com/example/nvdovin/weatherapp/utils/TemperatureScales.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.example.nvdovin.weatherapp.utils;
+
+public class TemperatureScales {
+    public static final int CELSIUS_SCALE = 273;
```
Comment (croitor-ivan) [inline app/src/main/java/com/example/nvdovin/weatherapp/utils/TemperatureScales.java]: wtf? parca am zsi fara constante.....

Comment (CebotariM) [reply app/src/main/java/com/example/nvdovin/weatherapp/utils/TemperatureScales.java]: Clasa va fi editata ulterior cand se va adauga Fahrenheit. Pana atunci am nevoie de valoarea data

File: app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    public void setLastUpdateTime() {
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putLong(LAST_UPDATE_TIME_KEY, System.currentTimeMillis());
+        editor.apply();
+    }
+
+    public boolean lastUpdateExceededLimit() {
+        if ((sharedPref.getLong(LAST_UPDATE_TIME_KEY, 0) == 0 ||
+                System.currentTimeMillis() - sharedPref.getLong(LAST_UPDATE_TIME_KEY, 0) > TimeUnit.HOURS.toMillis(TIME_SINCE_LAST_UPDATE))) {
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java]: `sharedPref.getLong(LAST_UPDATE_TIME_KEY, 0)` code duplicate, please replace it with local variable.

File: app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            setLastUpdateTime();
```
Comment (MaximCeban) [inline app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java]: why do you use this method here? you check time  and  refresh updated time, this is logically incorrect. You should use `setLastUpdatedTime()` in code after you update weather data. Remove it from here.

**Review by MaximCeban — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    public void setLastUpdateTime() {
+        SharedPreferences.Editor editor = sharedPref.edit();
+        editor.putLong(LAST_UPDATE_TIME_KEY, System.currentTimeMillis());
+        editor.apply();
+    }
+
+    public boolean lastUpdateExceededLimit() {
+        long getLastUpdateTime = sharedPref.getLong(LAST_UPDATE_TIME_KEY, 0);
+        if ((getLastUpdateTime == 0 ||
```
Comment (croitor-ivan) [inline app/src/main/java/com/example/nvdovin/weatherapp/utils/SharedPrefs.java]: ????
get rid of if, we don't need it.

**Pull request closed without merging**